src/main/java/com/microsoft/sqlserver/jdbc/ParameterMetaDataCache.java

```diff
@@ -156,12 +158,11 @@ static boolean addQueryMetadata(Parameter[] params, ArrayList<String> parameterN
                     cryptoCopy = new CryptoMetadata(metaData.getCekTableEntry(), metaData.getOrdinal(),
                             metaData.getEncryptionAlgorithmId(), metaData.getEncryptionAlgorithmName(),
                             metaData.getEncryptionType().getValue(), metaData.getNormalizationRuleVersion());
-                }
-                if (cryptoCopy != null && !cryptoCopy.isAlgorithmInitialized()) {
+                    if (cryptoCopy.isAlgorithmInitialized()) {
+                        return false;
+                    }
                     String paramName = parameterNames.get(i);
                     metadataMap.put(paramName, cryptoCopy);
-                } else {
-                    return false;
                 }
             } catch (SQLServerException e) {
                 MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_CryptoCacheInaccessible"));

```

Comment:
I'm thinking we should add more info to this error to help with debugging since I'm not sure if the exception would contain all the info we want.

Reply:
Both `R_CryptoCacheInaccessible` errors would originate from the same source: making a new CryptoMetaData with an invalid column encryption type, so I just passed that as an additional parameter to the final output.